### PR TITLE
RFC: Use service runner

### PR DESCRIPTION
**Pull request by gwicke**
This hooks up the new service runner / servisor and uses it to run restbase. Tests are passing, and a b/c compat server.js stub is provided so that the puppetization keeps working. We should be able to remove that later.

Before merging, lets discuss this properly.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -214,12 +219,14 @@ function setupConfigDefaults(conf) {
...
-    conf = setupConfigDefaults(conf);
+function main(options) {
+    var conf = setupConfigDefaults(options.config);
     // Set up the global options object with a logger
     var opts = {
-        log: rbUtil.makeLogger(conf.logging),
-        conf: conf
+        conf: conf,
+        logger: options.logger,
+        log: options.logger.log.bind(options.logger),
```
Comment (d00rman) [inline lib/server.js]: why are there both `log` and `logger` ?

File: package.json [hunk 2]
```diff
@@ -46,7 +47,8 @@
     "swagger-router": "^0.0.3",
     "swagger-ui": "git+https://github.com/wikimedia/swagger-ui#master",
     "url-template": "2.0.4",
-    "yargs": "~1.3.0"
+    "yargs": "~1.3.0",
+    "servisor": "git+https://github.com/gwicke/servisor#master"
```
Comment (d00rman) [inline package.json]: `/wikimedia/servisor`

File: lib/server.js (hunk 1 above)
Comment (gwicke) [inline lib/server.js]: The service runner code no longer sets up the magic function that also has a .child method attached to it. Instead, it exposes a plain logger object.

I'm torn about re-introducing the magic. It sure is more convenient for users, just feels a bit dirty.

File: package.json (hunk 2 above)
Comment (gwicke) [inline package.json]: Yup. Ideally, lets change straight to a real package once we have decided on a name.

File: test/utils/assert.js [hunk 3]
```diff
@@ -26,8 +26,8 @@ function localRequests(slice, expected) {
         }),
         expected,
         expected ?
-          'Should not have made remote request' :
-          'Should have made a remote request'
+          'Should not have made local request' :
+          'Should have made a local request'
```
Comment (d00rman) [inline test/utils/assert.js]: good catch :+1:

Comment (gwicke) [inline test/utils/assert.js]: Led me down the wrong path for an hour or two..

Comment (d00rman) [inline test/utils/assert.js]: Ups... Sorry about that :disappointed:

File: test/utils/server.js [hunk 4]
```diff
@@ -32,16 +39,13 @@ function start(_options) {
...
-            },
-            spec: config.spec
-        }).then(function(server){
+        console.log('starting restbase in '
+                + (options.offline ? 'OFFLINE' : 'ONLINE') + ' mode');
+        config.conf.offline = options.offline || false;
+
+        return runner.run(config.conf)
+        .then(function(servers){
+            var server = servers[0];
```
Comment (d00rman) [inline test/utils/server.js]: With servisor, ops, service-runner, this becomes really easy to read. Me likes mucho.

File: package.json (hunk 2 above)
Comment (d00rman) [inline package.json]: Yep. Upload the pkg first, though ;)

File: test/utils/server.js [hunk 5]
```diff
@@ -3,7 +3,7 @@
 // mocha defines to avoid JSHint breakage
 /* global describe, it, before, beforeEach, after, afterEach */
 
-var restbase  = require('../../lib/server.js');
+var ServiceRunner = require('servisor');
```
Comment (hardikj) [inline test/utils/server.js]: service-runner now ;)

File: package.json [hunk 6]
```diff
@@ -4,6 +4,7 @@
   "description": "REST storage and service dispatcher",
   "main": "server.js",
   "scripts": {
+    "start": "servisor",
```
Comment (hardikj) [inline package.json]: 'npm start' will be used to start servisor(now service-runner) or restbase?

Comment (d00rman) [inline package.json]: `npm start` will run `service-runner` which will read RESTBase's config and start it.

Comment (hardikj) [inline package.json]: Ah, nice!
We should modify the Readme as well with the above changes.

**Pull request merged**